src/bun.js/bindings/JSBuffer.cpp

```diff
@@ -250,25 +252,29 @@ const unsigned U32_MAX = std::numeric_limits<unsigned>().max();
 
 static inline uint32_t parseIndex(JSC::JSGlobalObject* lexicalGlobalObject, JSC::ThrowScope& scope, ASCIILiteral name, JSValue arg, size_t upperBound)
 {
-    if (!arg.isNumber()) return Bun::ERR::INVALID_ARG_TYPE(scope, lexicalGlobalObject, name, "number"_s, arg);
+    if (!arg.isNumber()) {
```

Comment:
can we check if it's an Int32 and avoid the cost of converting to a double for the most common case?